### DISCUSSION TITLE
[feature] Support for locally-run vector database

### DISCUSSION THREAD

**Discussion by ThreepE0** [pinecone-io/canopy#206] [Announcements]
Is local vector database support on the roadmap?  Looking for a privacy-focused rag solution, and needing to send all my notes to a cloud service is a deal-breaker for me.

**Comment by igiloh-pinecone** [comment]
Hi @ThreepE0 , thank you for reaching out!  

In the near future, we are planning to introduce in the retrieval related features that would assist users to get the most relevant results out of their RAG system. Those retrieval features would probably be tailored around Pinecone Database's built-in features. Those features might not be compatible with other vector DB.

Please note that Pinecone does have a secure and private offering, that is compliant with all SoC2, GDPR and HIPPA certifications. Feel free to reach out Pinecone support for more details.  
For now, I'm differing this issue to our [discussion board](https://github.com/pinecone-io/canopy/discussions), and we would love to hear more about use cases where this is required, and which solution you will be interested in.

**Comment by igiloh-pinecone** [comment]
Having said that, Canopy is an open source project, and we will of course respect community contributed PRs that would add support to any OSS vector database that is compliant with our testing and code quality standard.
